### DISCUSSION TITLE
How about adding a file upload button?

### DISCUSSION THREAD

**Discussion by ghost** [datenstrom/community#130] [Ask a question] [answered]
Hi,

Do you have any plans on integrating a simple file upload button in admin ?

**Comment by ghost** [comment]
This is how you doit... ; )
![load](https://cloud.githubusercontent.com/assets/6350636/11387320/b0a22818-9329-11e5-85f2-562a32ff31ad.gif)

**Comment by nibreh** [comment]
Hi

This is exactly what I suggested here https://github.com/datenstrom/yellow/issues/66#issuecomment-143311343

**Comment by ghost** [comment]
Yep cool huh !?
I´m using @Rovak : s  plugin https://github.com/Rovak/InlineAttachment , buttersmooth ; ) !! 
Maybe mr @markseu have time in the future to make a proper plugin, I just hardcoded the stuff for now.. haven´t have time to read the docs properly ; ) I´t might be some serious security issues, are no PHP:ninja ; )

**Comment by ghost** [comment]
hmm now all the files are uploaded to the media/images folder, might be better to upload it directly to the content/"yourpostfolder". In that way, when you delete the post the files will be deleted as well (i think ;)

**Comment by ghost** [comment]
something like this would be nice !

![art1](https://cloud.githubusercontent.com/assets/6350636/11399080/5844e6ea-9385-11e5-8e4c-545522bfaebb.png)
![art2](https://cloud.githubusercontent.com/assets/6350636/11399079/584558e6-9385-11e5-9437-ad0d4a25cdf0.png)
![art3](https://cloud.githubusercontent.com/assets/6350636/11399078/5844055e-9385-11e5-8b68-113b1fecc8b8.png)

**Comment by markseuffert** [comment]
Drag-and-drop file upload looks great. Any help in making Yellow better would be much appreciated. :smile:

**Comment by BsNoSi** [comment]
I love this way for simple adding files (thumb up). **BUT:**
- How can I handle i.e. PDF-files, that shall not be displayed, but available for download?
- How can I control the place, where the files are stored?
- How can I create directories?
- How are duplicate file names supervised? Gallery images will collide by same names like `011.jpg`. 

For simply adding a picture, this is really nice. But „only this“ is not enough for propper handling. I did not realy deeply think about it, but these requirements should be met in this case:
- dragging files will create _always_ a dir in /media/[filename], to keep added files together and protect against accidental overwriting
- depending to a list of file extentions (ini-file?) files are inserted as an image or as a download link. Last one maybe with a file type identifying icon. 
- There should be an option for uploading selected files additional to drag and drop, comparable to the method here (s. line below github edit field).

**Comment by redtide** [comment]
I agree with @BsNoSi and @nibreh 
It could be nice to have something like the simple-file-manager, it allows drag & drop and file handling.
It seems to need just some changes (setting the password in yellow editor/admin and add a rename file feature).

**Comment by punksinatra** [comment]
can the upload files be sorted automatically to refer to each post? I think it would be much easier to sort out files and everything instead of having them all in one directory

**Comment by xsinger** [comment]
* [Tiny PHP File Manager](https://github.com/prasathmani/tinyfilemanager)
* [PHP Filemanager](https://github.com/Den1xxx/Filemanager)
* [PHP File Manager](https://github.com/alexantr/filemanager)
* [php_srrFileManager](https://github.com/caos30/php_srrFileManager)
* [OneFileCMS](https://github.com/Self-Evident/OneFileCMS)

**Comment by markseuffert** [comment, marked as answer]
![screen capture](https://user-images.githubusercontent.com/2429165/39573416-69bdafa4-4ed3-11e8-94db-d1bc1767a4ed.gif)

I added simple file upload based on @nibreh’s work. Please give it a try.

**Comment by BsNoSi** [comment]
Is it a basic function of the editor or is a plugin required? I have tried it on Windows 10 pro but there is no upload. Only the image code is added.

**Comment by markseuffert** [comment]
File upload is part of the editor, no additional plugin needed anymore. I have tested it on Mac and Linux.

**Comment by BsNoSi** [comment]
Hm. I will test this with a few more browsers. Current status is, that Vivaldi and Firefox (both current version) apparently do not support this on Windows 10 (current version).

**Comment by schulle4u** [comment]
Can't confirm this. File upload is working for me with Firefox and Vivaldi under windows 10.
